Modules/_sqlite/statement.c

```diff
@@ -372,23 +372,36 @@ static void
 pysqlite_statement_dealloc(pysqlite_Statement *self)
 {
     PyTypeObject *tp = Py_TYPE(self);
+    PyObject_GC_UnTrack(self);
 
     if (self->st) {
         Py_BEGIN_ALLOW_THREADS
         sqlite3_finalize(self->st);
         Py_END_ALLOW_THREADS
+        self->st = NULL;
```

Comment:
No need, you are deallocating it already ;)

Reply:
Actually, we can remove the `if (self->st)` also; `sqlite3_finalize` is a no-op when passed a NULL pointer.

Reply:
OTOH, we can keep it as it is and move it to `stmt_clear()`; it is in practise the same as `Py_CLEAR` (free resources and zero out the member).

Reply:
I went for the latter, and I kept `self->st = NULL` to mimic `Py_CLEAR`; there's a possibility that the destructor calls back into code that tries to use this attribute again, right? Quoting the [SQLite docs](https://sqlite.org/c3ref/finalize.html):
_"It is a grievous error for the application to try to use a prepared statement after it has been finalized. Any use of a prepared statement after it has been finalized can result in undefined and undesirable behavior such as segfaults and heap corruption."_